### PR TITLE
docs: Documentation for Pause/Resume functionality.

### PR DESCRIPTION
**Pull request by jnh5y**
Docs for https://github.com/confluentinc/ksql/pull/9203

### CODE REVIEW

**Review by JimGalasyn — APPROVED** [review]
Awesome, thank you for the updates!

File: docs/developer-guide/ksqldb-reference/pause.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+layout: page
+title: PAUSE
+tagline:  ksqlDB PAUSE statement
+description: Syntax for the PAUSE statement in ksqlDB
+keywords: ksqlDB, query, end, stop
```
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-reference/pause.md:6]: These look like they should be updated? Same for `resume.md`

File: docs/developer-guide/ksqldb-reference/pause.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+--------
+
+```sql
+PAUSE query_id | ALL;
+```
+
+Description
+-----------
+
+Pause a persistent query.  Transient queries cannot be paused or resumed.
```
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-reference/pause.md:22]: Are there Kafka Streams docs detailing what it actually means to pause a query (e.g., if users have questions about whether their metrics will stop being emitted, whether restore will stop, etc)? If yes, it'd be good to link to those docs from here.

File: docs/operate-and-deploy/migrations-tool.md [hunk 3]
```diff
@@ -37,6 +37,8 @@ types of ksqlDB statements:
 * `CREATE TABLE ... AS SELECT`
 * `CREATE OR REPLACE`
 * `INSERT INTO ... AS SELECT`
+* `PAUSE <queryID>`
```
Comment (vcrfxia) [inline docs/operate-and-deploy/migrations-tool.md:40]: Can we also update the java client docs at https://github.com/confluentinc/ksql/blob/0d973038399b95006c3334800fb31ffe587d4a94/ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/Client.java#L135-L136 (and for the other method overload)?

Comment (vcrfxia) [reply docs/operate-and-deploy/migrations-tool.md:40]: Also the java client docs (https://github.com/confluentinc/ksql/blob/master/docs/developer-guide/ksqldb-clients/java-client.md) describing what you can do with `executeStatement()`.

**Review by vcrfxia — COMMENTED** [review]
Thanks @jnh5y -- comments inline!

File: docs/developer-guide/ksqldb-reference/pause.md (hunk 2 above)
Comment (vcrfxia) [reply docs/developer-guide/ksqldb-reference/pause.md:22]: Do you have a link to the AK docs? I suspect that's more suitable than the KIP.

**Review by vcrfxia — APPROVED** [review]
LGTM besides the comment about adding more details about what it means to pause a query. Can be added after the fact if AK docs aren't available yet.

File: docs/developer-guide/ksqldb-reference/pause.md (hunk 2 above)
Comment (vcrfxia) [reply docs/developer-guide/ksqldb-reference/pause.md:22]: That's too bad. I can easily see users having all sorts of questions about it actually means to pause a query, and it'd be great if the docs answered them preemptively, but I don't think it's a blocker for getting this docs PR merged. 

I suspect you'll want to write a blurb for the release blog post (whenever this release goes out) about this feature anyway, so you can always augment these docs with similar content at that time.

**Comment by jnh5y** [comment]
@JimGalasyn since I just merged this PR, do I need to cherry-pick it to an 0.28.x branch?

**Comment by vcrfxia** [comment]
> since I just merged this PR, do I need to cherry-pick it to an 0.28.x branch?

@jnh5y yes.

**Comment by jnh5y** [comment]
Which one?  Is it `0.28` or the latest of the `cc-docker-ksql` ones?

```jhughes@C02GN02C1PG2 ksql % git branch -a | grep \\\.28
  remotes/origin/0.28
  remotes/origin/0.28-cleanup
  remotes/origin/0.28.0-cc-docker-ksql.110.x
  remotes/origin/0.28.1-cc-docker-ksql.114.x
  remotes/origin/0.28.1-cc-docker-ksql.115.x
  remotes/origin/0.28.2-cc-docker-ksql.116.x
  remotes/origin/0.28.2-cc-docker-ksql.117.x
  remotes/origin/0.28.2-cc-docker-ksql.119.x```

**Comment by vcrfxia** [comment]
`remotes/origin/0.28.2-cc-docker-ksql.119.x` -- it's linked from the spreadsheet where we're tracking the 0.28 release.

**Comment by jnh5y** [comment]
Cool, done.  FYI @lihaosky I tossed my docs PR into that branch.

**Pull request merged**